sklearn/datasets/_twenty_newsgroups.py

```diff
@@ -467,10 +488,24 @@ def fetch_20newsgroups_vectorized(*, subset="train", remove=(), data_home=None,
     with open(join(module_path, 'descr', 'twenty_newsgroups.rst')) as rst_file:
         fdescr = rst_file.read()
 
+    X = data
+    y = target
+    frame = None
+    target_name = ['Category_class',]
+
+    if as_frame:
+        frame, X, y = _convert_data_dataframe("fetch_20newsgroups_vectorized",
+                                              data.toarray(),
```

Comment:
This may need to be treated differently and use use `pd.DataFrame.sparse.from_spmatrix` to get a dataframe using the sparse extension arrays.

Reply:
oh, ok. this was the workaround we could come up, but it's good to know there is a more proper way to do the conversion.